core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -15,8 +15,10 @@ function taxonomy_overview_vocabularies($form) {
   $vocabularies = taxonomy_vocabulary_load_multiple(FALSE);
   $form['#tree'] = TRUE;
   foreach ($vocabularies as $vocabulary) {
+    $terms = db_query("SELECT count(tid) FROM {taxonomy_term_data} WHERE vocabulary = :vocab", array(':vocab' => $vocabulary->machine_name))->fetchField();
```

Comment:
As a noice dev, I find `db_select()` much easier to understand than direct db queries.

```suggestion
    $term_count = db_select('taxonomy_term_data', 'ttd')
      ->fields('ttd', array('tid'))
      ->condition('vocabulary', $vocabulary->machine_name)
      ->execute()
      ->rowCount();
```

Reply:
No, db_select is not only much slower, but it's harder for anyone who's done other web development before to understand. It only exists in the world of Drupal/Backdrop, but SQL queries exist for anyone using a SQL database anywhere. `db_select()` is likely to be removed in Backdrop 2.0 (or whenever) so we should be moving away from using these functions as much as possible.

Reply:
Fair enough 👍 ...perhaps still a good idea to rename the variable to `$term_count` though 🙂:

```suggestion
    $term_count = db_query("SELECT count(tid) FROM {taxonomy_term_data} WHERE vocabulary = :vocab", array(':vocab' => $vocabulary->machine_name))->fetchField();
```

Reply:
...btw, have you tried `->rowCount();` instead of `->fetchField();`? I have read that it is more performant, because it does not need to load the fields.